### PR TITLE
cylc authentication: don't report old client fails

### LINKED ISSUE #1578: cylc 6.6.0 suite logs can get swarmed by old-client messages
If you run a cylc 6.6.0 suite on a machine with other users running an older version of `cylc scan`, your log can be swarmed by large number of warning messages.

I believe the message is being logged by:
https://github.com/cylc/cylc/blob/master/lib/cylc/network/connection_validator.py#L87

### PR DESCRIPTION
**Pull request by benfitzpatrick**
This fixes #1578.

@matthewrmshin, please review 1.
@hjoliver, please review 2.

### CODE REVIEW

**Comment by matthewrmshin** [comment]
Looks OK 1.

File: lib/cylc/network/connection_validator.py [hunk 1]
```diff
@@ -67,7 +68,7 @@ def acceptIdentification(self, daemon, connection, token, challenge):
             # Back compat for old suite client (passphrase only)
             # (Allows old scan to see new suites.)
             proc_passwd = token
-            user = host = uuid = prog_name = "(old-client)"
+            is_old_daemon = True
```
Comment (hjoliver) [inline lib/cylc/network/connection_validator.py]: Should be `is_old_client`!

**Comment by hjoliver** [comment]
Review 2: good, but one misleading variable name as commented above.

**Comment by benfitzpatrick** [comment]
Done

**Pull request merged**